spec/behavior/url.spec.js

```diff
@@ -104,4 +104,30 @@ describe( "URL Template", function() {
 			href.should.equal( expected );
 		} );
 	} );
+
+	describe( "Timing token replacement", function() {
+		var testUrl = "/parent/:id/child/:child.id";
+
+		describe( "Replacing 2 tokens 1000 times with regex", function() {
+			var elapsed;
+			var urls = [];
+			before( function() {
+				var start = Date.now();
+				var tokens = url.getTokens( testUrl );
+				var halUrl = url.forHal( testUrl );
+				for ( var i = 0; i < 1000; i++ ) {
+					urls.push( url.process( _.clone( tokens ), halUrl, { id: 1, childId: 2 }, "parent" ) );
+				}
+				elapsed = Date.now() - start;
+			} );
+
+			it( "should produce a valid URL", function() {
+				urls[ 1 ].should.equal( "/parent/1/child/2" );
+			} );
+
+			it( "should be \"quick\"", function() {
+				elapsed.should.be.below( 40 );
```

Comment:
This one is failing for me about half the time. Never way above 40, though. I think the worst I've seen is 43.